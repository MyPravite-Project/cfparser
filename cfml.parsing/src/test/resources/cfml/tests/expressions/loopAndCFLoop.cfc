cfloop(query=SiteRelatedTables)
{
    loopy();
}

loop(query=SiteRelatedTables)
{
    loopy();
}