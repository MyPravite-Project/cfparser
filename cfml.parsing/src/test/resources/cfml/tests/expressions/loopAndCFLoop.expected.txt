/*===TOKENS===*/
LOOP                     <cfloop>
'('                      <(>
QUERY                    <query>
'='                      <=>
IDENTIFIER               <SiteRelatedTables>
')'                      <)>
'{'                      <{>
IDENTIFIER               <loopy>
'('                      <(>
')'                      <)>
';'                      <;>
'}'                      <}>
LOOP                     <loop>
'('                      <(>
QUERY                    <query>
'='                      <=>
IDENTIFIER               <SiteRelatedTables>
')'                      <)>
'{'                      <{>
IDENTIFIER               <loopy>
'('                      <(>
')'                      <)>
';'                      <;>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (element
    (statement
      (tagOperatorStatement
        (tagFunctionStatement
          (cfmlFunction cfloop)
          (
          (parameterList
            (parameter
              (identifier (cfmlFunction query))
              =
              (startExpression
                (compareExpression
                  (baseExpression
                    (unaryExpression (memberExpression (identifier SiteRelatedTables)))
                  )
                )
              )
            )
          )
          )
          (compoundStatement
            {
            (statement
              (startExpression
                (compareExpression
                  (baseExpression
                    (unaryExpression
                      (memberExpression
                        (functionCall (identifierOrReservedWord (identifier loopy)) ( argumentList ))
                      )
                    )
                  )
                )
              )
              ;
            )
            }
          )
        )
      )
    )
  )
  (element
    (statement
      (tagOperatorStatement
        (tagFunctionStatement
          (cfmlFunction loop)
          (
          (parameterList
            (parameter
              (identifier (cfmlFunction query))
              =
              (startExpression
                (compareExpression
                  (baseExpression
                    (unaryExpression (memberExpression (identifier SiteRelatedTables)))
                  )
                )
              )
            )
          )
          )
          (compoundStatement
            {
            (statement
              (startExpression
                (compareExpression
                  (baseExpression
                    (unaryExpression
                      (memberExpression
                        (functionCall (identifierOrReservedWord (identifier loopy)) ( argumentList ))
                      )
                    )
                  )
                )
              )
              ;
            )
            }
          )
        )
      )
    )
  )
)
/*======*/